### ISSUE TITLE
Servo Cannot Run html5test.com Test

### ISSUE THREAD

**Issue by sciguyryan** [servo/servo#17730]
Hi guys.

When attempting to run [html5test.com](http://html5test.com) Servo appears to stall and is unable to run any of the tests. This appears to be a regression as I was able to run this test before. The last time I had tested this was on the nightly from the 31st of May 2017.

This was tested on Windows 10 x64.

**Comment by fabricedesre** [comment]
I was able run the test on linux (Servo scored 259/555) so this looks like something OS specific.

**Comment by sciguyryan** [comment]
That's pretty interesting. I did attempt to get a simplified testcase written but no luck with that yet unfortunately.

**Comment by atouchet** [comment]
I just tested this in Windows 10 and got the same results as @fabricedesre. Doesn't seem to be due to the OS.

**Comment by jdm** [comment]
I got endless loading on macOS in a debug build.

**Comment by sciguyryan** [comment]
@jdm Yeah, that's what I got too but I'm no longer seeing that. Some intermittent issue perhaps?

**Comment by lumiscosity** [comment]
I don't see any issues on a dev build anymore. 381/555 with experimental features off, 393 with experimental features on!

Tested on Arch Linux x86_64, Servo built against commit 4fb71cc17e7faa6abb3b92b9240765c0ce1a985d, dev build.

<img width="1024" height="817" alt="Image" src="https://github.com/user-attachments/assets/cd2146ad-5993-409c-aad7-7d09fdfd9a40" />

**Comment by lumiscosity** [comment]
On second thought, I'd like to see this tested on Mac before this is closed, since it was confirmed at least sometimes working on Win10 and Linux.

**Comment by jdm** [comment]
Tested on macOS and it runs fine.
